src/game.cpp

```diff
@@ -1862,6 +1861,27 @@ static void update_faction_api( npc *guy )
     }
 }
 
+void game::validate_mounted_npcs()
+{
+    for( monster &m : all_monsters() ) {
+        if( m.has_effect( effect_ridden ) && m.mounted_player_id.is_valid() ) {
+            player *mounted_pl = g->critter_by_id<player>( m.mounted_player_id );
+            if( mounted_pl == nullptr ) {
+                // Target no longer valid.
+                m.mounted_player_id = character_id();
+                m.remove_effect( effect_ridden );
+                continue;
+            }
+            if( !mounted_pl->is_mounted() && mounted_pl->is_npc() ) {
+                mounted_pl->mounted_creature = critter_tracker->from_temporary_id(
+                                                   mounted_pl->mounted_creature_id );
```

Comment:
Shouldn't that return the very same mounted creature that you already have in `m`? If you do this just to get a `shared_ptr`, you can use `critter_tracker->find( m.pos() )` instead (`find` returns a `shared_ptr`).

Reply:
The monster only stores `mounted_player_id` through save/load , there wont be any `mounted_player` attached to the monster yet, this is to add it.

NPCs also store their mounted_creature as an id through save/load

Reply:
Oh I see what you mean, took me a while lol